### PR TITLE
Cmc integration

### PR DESCRIPTION
**Pull request by clydeautin**
## Type of Change

- [ ] Debugging
- [x] New Feature
- [ ] Refactor/File Clean-up

## Describe changes
create facade for coinmarketcap API consumption and also wrote test for happy and sad path of API.
Stored the API key in .env and gitignored it
## Checklist

- [x] I have made RSpec test  
- [x] My code is passing all tests 
- [x] I assigned a review for my PR

### CODE REVIEW

File: facades/cmc_facade.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+def get_btc_price():
+  url = 'https://pro-api.coinmarketcap.com/v2/cryptocurrency/quotes/latest'
+
+  parameters = {
+    'symbol': 'BTC'
+  }
+
+  headers = {
+    'Accepts': 'application/json',
+    'X-CMC_PRO_API_KEY': os.getenv('CMC_API_KEY')
```
Comment (jyaaan) [inline facades/cmc_facade.py:18]: Retrieve API key separately and validate it, raising if it's missing. Don't consume an API call if we don't have the proper credentials.

File: facades/cmc_facade.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  try: 
+    response = requests.get(url, params=parameters, headers=headers)
+    response.raise_for_status() # Raise an HTTPError for bad responses
+    price = response.json()['data']['BTC'][0]['quote']['USD']['price']
```
Comment (jyaaan) [inline facades/cmc_facade.py:23]: consider adding some data validation so we don't end up with `KeyError` if we experience and issue with the API
```python
data = response.json()
if 'data' not in data or 'BTC' not in data['data']:
    raise ValueError("Unable to validate response structure")
```

File: facades/cmc_facade.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import requests
+import locale
```
Comment (jyaaan) [inline facades/cmc_facade.py:2]: remove unused imports

File: tests/cmc_facade_test.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import pytest
+import requests
+from unittest.mock import patch
```
Comment (jyaaan) [inline tests/cmc_facade_test.py:3]: in the future, `MagicMock` is a better way to manage your mock responses.

File: tests/cmc_facade_test.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+                        "price": 45000.00
+                    }
+                }
+            }]
+        }
+    }
+
+def test_get_btc_price_success(mock_btc_response):
+    with patch('requests.get') as mock_get:
+        mock_get.return_value.json.return_value = mock_btc_response
```
Comment (jyaaan) [inline tests/cmc_facade_test.py:22]: I think it's also important to test for status codes when developing callouts. 
`mock_get.return_value.status_code = 200` would be good.

File: tests/cmc_facade_test.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        price = get_btc_price()
+        assert price == 45000.00
+        mock_get.assert_called_once()
```
Comment (jyaaan) [inline tests/cmc_facade_test.py:25]: change to `mock_get.return_value.raise_for_status.assert_called_once()` to test for status code.

File: tests/cmc_facade_test.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+def test_get_btc_price_api_error():
+    with patch('facades.cmc_facade.requests.get') as mock_get:
+        mock_get.side_effect = requests.exceptions.RequestException("API Error")
+        with pytest.raises(Exception) as exc_info:
+            get_btc_price()
+        assert "Failed to fetch BTC price" in str(exc_info.value)
```
Comment (jyaaan) [inline tests/cmc_facade_test.py:32]: Add test for invalid response. can set return value to some garbage dict like `{'this_is': 'bad_response'}`

**Review by jyaaan — COMMENTED** [review]
looks good! this should work for our use case, but some improvements can be made. for the sake of time, i'm just commenting so you can merge immediately.

File: facades/cmc_facade.py (hunk 2 above)
Comment (jyaaan) [reply facades/cmc_facade.py:23]: you could also catch key or value errors in an exception too.

**Comment by clydeautin** [comment]
Going to merge this now and then do another PR that will incorporate your recommendations

**Pull request merged**